otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -399,7 +402,13 @@ def hook(module, inp, outp):
             if raw_model.__class__.__name__ == "NNCFNetwork":
                 raw_model = raw_model.get_nncf_wrapped_model()
             if isinstance(raw_model, TwoStageDetector):
-                saliency_hook = ActivationMapHook(feature_model)
+                test_pipeline = cfg.data.test.pipeline
+                width, height = None, None
+                for pipeline in test_pipeline:
+                    width, height = pipeline.get("img_scale", (None, None))
+                if height is None:
+                    raise ValueError("img_scale has to be defined in the test pipeline.")
+                saliency_hook = MaskRCNNHook(feature_model, input_img_shape=(height, width))
```

Comment:
Will we use `MaskRCNNHook` for all of the two-stage detection models? If so, this name should be changed.

Reply:
`MaskRCNNRecordingForwardHook` is designed for MaskRCNN. It will not work with other two-stage detector.
But, afaik so far we do not have plans to support any other two-stage detector rather than m-rcnn, so, to keep it simple and clear, I propose to keep it as is (rather that support two cases: 1. maskrcnn and 2. some other general two-stage detector). What do you think?